lib/property_builder.js

```diff
@@ -40,6 +40,7 @@ function PropertyBuilder (propName, strict) {
   Property.prototype.required = function (required) { return this.set('required', ((required === false) ? false : true)); }
   Property.prototype.default  = function (val) { return this.set('default', val); }
   Property.prototype.unique   = function (val) { return this.index('unique', ((val === false) ? false : true)); }
+  Property.prototype.ref      = function (modelName) { return this.set('ref', modelName); }
```

Comment:
I'm fine with this pattern, but if you are going to add this, then we should remove schema (or at least mark it deprecated)

Also this does not solve the array of objects.  I was looking at this for a while last night and figured a few things out but we can't implement it without some minor changes.


According to mongoose: http://mongoosejs.com/docs/populate.html
Defining multiples (think players on team)
```[{ type: Schema.Types.ObjectId, ref: 'player' }]```

My initial result was:
```
Property.prototype.schemas  = function (ref, type) { 
    var thisType = type ? type : Schema.Types.ObjectId;
    this.data = [ { ref : ref, type : thisType } ];
    return this; 
  }
```

It sucks because it doesnt match the pattern of the other object by using ```set```.

Furthermore it will require a change to fin()
```
  Property.prototype.fin = function () {
    //If the data has a type or is an array
    **if (!( this.data && ( this.data.type || Array.isArray(this.data) ) ) ) {**
      throw new Error('No type set for schema property ' + propName);
    } else {
      return this.data;
    }
  }
```

But that is a pretty big change